### PR TITLE
docs: Cleanup more literals in API docs

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/root/configuration/http/http_filters/aws_lambda_filter.rst [hunk 1]
```diff
@@ -22,7 +22,8 @@ If :ref:`payload_passthrough <envoy_v3_api_field_extensions.filters.http.aws_lam
 However, if :ref:`payload_passthrough <envoy_v3_api_field_extensions.filters.http.aws_lambda.v3.Config.payload_passthrough>`
 is set to ``false``, then the HTTP request is transformed to a JSON payload with the following schema:
 
-.. code-block::
+.. code-block:: json
+   :force:
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/aws_lambda_filter.rst:26]: requires css hack

File: api/envoy/config/bootstrap/v3/bootstrap.proto [hunk 2]
```diff
@@ -267,12 +267,12 @@ message Bootstrap {
...
-  // can be packed into this *typed_dns_resolver_config*. This configuration replaces the
+  // can be packed into this ``typed_dns_resolver_config``. This configuration replaces the
   // :ref:`dns_resolution_config <envoy_v3_api_field_config.bootstrap.v3.Bootstrap.dns_resolution_config>`
   // configuration.
-  // During the transition period when both *dns_resolution_config* and *typed_dns_resolver_config* exists,
-  // when *typed_dns_resolver_config* is in place, Envoy will use it and ignore *dns_resolution_config*.
-  // When *typed_dns_resolver_config* is missing, the default behavior is in place.
+  // During the transition period when both ``dns_resolution_config`` and ``typed_dns_resolver_config`` exists,
+  // when ``typed_dns_resolver_config`` is in place, Envoy will use it and ignore ``dns_resolution_config``.
+  // When ``typed_dns_resolver_config`` is missing, the default behavior is in place.
```
Comment (htuch) [inline api/envoy/config/bootstrap/v3/bootstrap.proto:275]: Is this covered by the RST warnings? Is there a style guide entry here?

File: docs/root/configuration/http/http_filters/aws_lambda_filter.rst (hunk 1 above)
Comment (htuch) [reply docs/root/configuration/http/http_filters/aws_lambda_filter.rst:26]: Can you unbundle the changes with interesting things from the `**` --> quotes mass change. Hard to spot what to pay attention to.

Comment (phlax) [reply docs/root/configuration/http/http_filters/aws_lambda_filter.rst:26]: yep, that was kinda my intention

File: api/envoy/config/bootstrap/v3/bootstrap.proto (hunk 2 above)
Comment (phlax) [reply api/envoy/config/bootstrap/v3/bootstrap.proto:275]: how could it - this is me editing prose to make words that should be literal, so - until we get AI in the pipeline this is not something we can test for

more generally - the use of `*` for literals relates to historic advice which i changed long ago

we could perhaps check for `*word*` and spellcheck the word as literals should not be marked up this way - it would catch some - ill look at it when i get to the spellcheck stuff

File: docs/root/configuration/http/http_filters/aws_lambda_filter.rst (hunk 1 above)
Comment (phlax) [reply docs/root/configuration/http/http_filters/aws_lambda_filter.rst]: code changes are here and here:

- #22198 
- #22197

File: api/envoy/config/bootstrap/v3/bootstrap.proto (hunk 2 above)
Comment (adisuissa) [reply api/envoy/config/bootstrap/v3/bootstrap.proto:275]: I guess an alternative would be to clarify it in https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md, but as long as we don't detect this in tooling, it will be hard to catch.
Can we hook a validation to the comments parser somehow?

Comment (phlax) [reply api/envoy/config/bootstrap/v3/bootstrap.proto:275]: this is what we do now for single backticks

for `*` or bare literals its much harder to detect - there are genuine uses of `*` - ie for italics/emphasis - and there is nothing to grep for with bare literals

spellchecking could work and catch most issues but not all and with some false +ves etc

this is something i intend to work, but not imminently - some preparatory work has been done over in pytooling tho

wrt to the STYLE guide i can update/review if you think it needs it but would suggest we follow up with that

Comment (phlax) [reply api/envoy/config/bootstrap/v3/bootstrap.proto:275]: also - mho is that the best style guide is what is there - the more we clean stuff up the more people are working from examples that exemplify what we want

Comment (adisuissa) [reply api/envoy/config/bootstrap/v3/bootstrap.proto:275]: > also - mho is that the best style guide is what is there - the more we clean stuff up the more people are working from examples that exemplify what we want

I also tend to follow other examples.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
@htuch for a final say.

**Comment by phlax** [comment]
one thing im thinking - we could do with a "literals" policy (myabe that should be `literals`)

my rule of thumb is "would this be localized" - but that doesnt always help

there are some grey areas, like HTTP or `HTTP` - which on the basis of it being more acronym than literal it probably should not be a literal - but im not sure if that is ever localized any other way (checking eg https://pontoon.mozilla.org/machinery/) 

tldr - i think we just need to agree some rules/priorites regarding literals and code blocks and put those in the style guide - but lets follow up with that

**Review by htuch — APPROVED** [review]
Thanks for the cleanup. I do think it would be ideal to add at least a style guide entry in a followup, since this is the kind of thing that will regress. Examples are powerful, but some guidance is even better :P
